nirc_ehr/resources/queries/study/cases.js

```diff
@@ -15,8 +15,14 @@ function onInit(event, helper) {
 
 EHR.Server.TriggerManager.registerHandlerForQuery(EHR.Server.TriggerManager.Events.BEFORE_UPSERT, 'study', 'cases', function(helper, errors, row, oldRow){
     if (!helper.isETL()) {
-        if (row.enddate && !triggerHelper.canCloseCase(row.category)) {
+        if (row.enddate) {
+            if(!triggerHelper.canCloseCase(row.category)) {
                 EHR.Server.Utils.addError(errors, 'enddate', 'Veterinarian permission required to close a case.', 'ERROR');
+            }
+
+            if (!row.closeRemark) {
```

Comment:
may be having this check before canCloseCase avoids the db query during canCloseCase?

Reply:
yeah that makes sense to speed up the normal use case